shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-spring/shardingsphere-sharding-spring-namespace/src/test/java/org/apache/shardingsphere/sharding/spring/namespace/ShardingSpringNamespaceTest.java

```diff
@@ -163,26 +185,70 @@ public void assertOrderKeyGenerator() {
+        Collection<ShardingTableRuleConfiguration> actualComplexRuleConfigurations = complexRule.getTables();
+        assertFalse(actualComplexRuleConfigurations.isEmpty());
+        assertThat(actualComplexRuleConfigurations.size(), is(1));
+        ShardingTableRuleConfiguration actualComplexRuleConfiguration = actualComplexRuleConfigurations.iterator().next();
+        assertThat(actualComplexRuleConfiguration.getLogicTable(), is("t_order"));
+        assertThat(actualComplexRuleConfiguration.getActualDataNodes(), is("ds_$->{0..1}.t_order_$->{0..3}"));
+        assertThat(actualComplexRuleConfiguration.getDatabaseShardingStrategy().getShardingAlgorithmName(), is("dataSourceShardingAlgorithm"));
+        assertThat(actualComplexRuleConfiguration.getTableShardingStrategy().getShardingAlgorithmName(), is("orderTableShardingAlgorithm"));
+        assertThat(actualComplexRuleConfiguration.getKeyGenerateStrategy().getKeyGeneratorName(), is("incrementAlgorithm"));
+        assertThat(complexRule.getDefaultKeyGenerateStrategy().getKeyGeneratorName(), is("incrementAlgorithm"));
+        
     }
     
     @Test
     public void assertBindingRule() {
-        // TODO
+        Collection<ShardingTableRuleConfiguration> actualBindingTableRuleConfigurations = bindingRule.getTables();
+        assertFalse(actualBindingTableRuleConfigurations.isEmpty());
+        assertThat(actualBindingTableRuleConfigurations.size(), is(4));
+        Iterator<ShardingTableRuleConfiguration> actualIterator = actualBindingTableRuleConfigurations.iterator();
+        assertThat(actualIterator.next().getLogicTable(), is("t_order"));
+        assertThat(actualIterator.next().getLogicTable(), is("t_order_item"));
+        assertThat(actualIterator.next().getLogicTable(), is("t_user"));
+        assertThat(actualIterator.next().getLogicTable(), is("t_user_detail"));
+        Collection<String> actualBindingTableGroups = bindingRule.getBindingTableGroups();
+        assertFalse(actualBindingTableGroups.isEmpty());
+        assertThat(actualBindingTableGroups.size(), is(2));
+        assertTrue(actualBindingTableGroups.containsAll(Arrays.asList("t_order, t_order_item", "t_order, t_order_item")));
     }
     
     @Test
     public void assertBroadcastRule() {
-        // TODO
+        Collection<ShardingTableRuleConfiguration> actualBroadcastTableConfigurations = broadcastRule.getTables();
+        assertFalse(actualBroadcastTableConfigurations.isEmpty());
+        assertThat(actualBroadcastTableConfigurations.size(), is(2));
+        Iterator<ShardingTableRuleConfiguration> actualIterator = actualBroadcastTableConfigurations.iterator();
+        assertThat(actualIterator.next().getLogicTable(), is("t_order"));
+        assertThat(actualIterator.next().getLogicTable(), is("t_order_item"));
+        Collection<String> broadcastTables = broadcastRule.getBroadcastTables();
+        assertFalse(broadcastTables.isEmpty());
+        assertThat(broadcastTables.size(), is(2));
```

Comment:
It is unnecessary for assert both `broadcastTables.isEmpty()` and assert `size is 2`

Reply:
fixed